analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInitializerOrder.cs

```diff
@@ -118,16 +112,15 @@ private static List<IdentifierTypeDeclarationMapping> GetIdentifierTypeMappings(
                     return new IdentifierFieldMapping
                     {
                         Field = field,
-                        IsRelevant = field != null &&
-                            !field.IsConst &&
-                            field.IsStatic &&
-                            containingType.Equals(field.ContainingType) &&
-                            semanticModel.GetEnclosingSymbol(identifier.SpanStart) is IFieldSymbol enclosingSymbol &&
-                            enclosingSymbol.ContainingType.Equals(field.ContainingType)
+                        IsRelevant = field != null
```

Comment:
It's a thing of taste, but it could be optimized to
_field is {IsConst: false, IsStatic: true}_